### ISSUE TITLE
Reduce macro spam in docs

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#2108]
## How can Bevy's documentation be improved?

Several of our macro-generated trait impls (e.g. [WorldQuery](https://docs.rs/bevy/0.5.0/bevy/ecs/query/trait.WorldQuery.html)) are flooded with near duplicate implementations that make it hard to browse. This is a workaround the lack of variadic generics, but we should try to reduce the end user impact.

We should be able to hide this with `#[doc(hidden)]`. Ideally, we get clever with our macro and hide all-but-one of the impls of each relevant type.

**Comment by mockersf** [comment]
Unless rustdoc can display them in a nicer way,I would rather keep them as is in the doc and not hide them

**Comment by alice-i-cecile** [comment]
What if we displayed only one and then commented that equivalent impls exist for every combination up to 16?

**Comment by mockersf** [comment]
I would prefer we don't, and keep our doc the same as Rust for that ([for example](https://doc.rust-lang.org/std/primitive.tuple.html#impl-Debug)).

Additionally I don't think it's an issue as I like having them all, but I could imagine improvements at rustdoc level rather than at each doc item

**Comment by ahmedcharles** [comment]
Perhaps this shouldn't be marked as a good first issue if the plan is to make rustdoc changes. That's unlikely to be a good first bevy issue (even if it would be a good first rustdoc issue).

**Comment by cart** [comment]
Yup I agree that this is currently suboptimal, but I also don't want to hide the details of the implementation from people. Variadics would be 10/10 cool, but given the unlikelihood of us getting those any time soon, I agree that trying to improve the presentation in upstream rustdoc seems like the move. I'm happy to leave this issue open until we find a solid solution.
